features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/processor/EnrichedResource.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.processor
+
+import com.google.gson.JsonObject
+
+internal data class EnrichedResource(
+    internal val resource: ByteArray,
+    internal val applicationId: String,
+    internal val filename: String,
+    internal val metadata: JsonObject = JsonObject().apply {
+        this.addProperty(APPLICATION_ID_KEY, applicationId)
+        this.addProperty(FILENAME_KEY, filename)
+    }
```

Comment:
I think it is better to avoid bundling something for the storage of the class with a class itself. I would propose rather create some extension method which will return metadata needed as a JSON on demand.